connectors/mysql.go

```diff
@@ -0,0 +1,261 @@
+package connectors
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/LUSHDigital/modelgen/model"
+	"github.com/LUSHDigital/modelgen/sqlfmt"
+	"github.com/LUSHDigital/modelgen/templates"
+)
+
+type mySQLExplain struct {
+	Field   *string
+	Type    *string
+	Null    *string
+	Key     *string
+	Default *string
+	Extra   *string
+}
+
+var mysqlDataTypes = map[string]model.SQLType{
+	"char":    model.TypeString,
+	"varchar": model.TypeString,
+
+	"tinytext":   model.TypeString,
+	"text":       model.TypeString,
+	"mediumtext": model.TypeString,
+	"longtext":   model.TypeString,
+
+	"json": model.TypeJSON,
+
+	"enum": model.TypeString,
+	"set":  model.TypeString,
+
+	"bit":        model.TypeByteSlice,
+	"binary":     model.TypeByteSlice,
+	"varbinary":  model.TypeByteSlice,
+	"tinyblob":   model.TypeByteSlice,
+	"mediumblob": model.TypeByteSlice,
+	"blob":       model.TypeByteSlice,
+	"longblob":   model.TypeByteSlice,
+
+	"tinyint_as_bool": model.TypeBool,
+	"tinyint":         model.TypeInt,
+	"smallint":        model.TypeInt,
+	"mediumint":       model.TypeInt,
+	"int":             model.TypeInt,
+	"bigint":          model.TypeInt,
+
+	"float":   model.TypeFloat,
+	"double":  model.TypeFloat,
+	"decimal": model.TypeFloat,
+
+	"date":      model.TypeDate,
+	"datetime":  model.TypeDate,
+	"timestamp": model.TypeDate,
+	"time":      model.TypeString,
+	"year":      model.TypeInt,
+}
+
+type MySQL struct {
+	username string
+	password string
+	host     string
+	port     string
+	database string
+}
+
+func NewMySQL(username, password, host, port, database string) *MySQL {
+	return &MySQL{
+		username: username,
+		password: password,
+		host:     host,
+		port:     port,
+		database: database,
+	}
+}
+
+func (m *MySQL) Connect() (*sql.DB, error) {
```

Comment:
Missing documentation. There are many other instances, it's minor but it would be a marked improvement if we could add it.